src/components/VmDetails/cards/NicsCard/NicEditor.js

```diff
@@ -0,0 +1,303 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import {
+  Button,
+  Col,
+  ControlLabel,
+  FieldLevelHelp,
+  Form,
+  FormControl,
+  FormGroup,
+  Modal,
+  Radio,
+} from 'patternfly-react'
+import SelectBox from '../../../SelectBox'
+import ExpandCollapseSection from '../../../ExpandCollapseSection'
+import NicLinkStateIcon from './NicLinkStateIcon'
+
+import { msg } from '../../../../intl'
+import style from './style.css'
+
+const EMPTY_ID = ''
+
+const NIC_INTERFACES = [
+  {
+    id: 'rtl8139',
+    value: 'rtl8139',
+  },
+  {
+    id: 'e1000',
+    value: 'e1000',
+  },
+  {
+    id: 'virtio',
+    value: 'VirtIO',
+  },
+]
+const NIC_INTERFACE_DEFAULT = 'virtio'
+const NIC_INTERFACE_CANT_CHANGE = [ 'pci_passthrough' ]
+
+const LabelCol = ({ children, ...rest }) => {
+  return <Col componentClass={ControlLabel} {...rest}>
+    { children }
+  </Col>
+}
+LabelCol.propTypes = {
+  children: PropTypes.node.isRequired,
+}
+
+/**
+ * Collect required information required to create a new NIC for a VM.
+ *
+ * Required data:
+ *   - Name (auto name the next number higher?)
+ *   - VNIC Profile (which includes the network) <-- needs to be on same DC and the VM
+ *
+ * Optional data:
+ *   - Network Type (dual mode virtio and rtl8139, virtio, rtl8139, e1000)
+ *     - NOTE: can only be changed if VM is down or Nic (card) is unplugged
+ *   - Link State (Up = connected, Down = cable unplugged)
+ */
+class NicEditor extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      showModal: false,
+
+      id: undefined,
+      name: props.nextNicName,
+      vnicProfileId: EMPTY_ID,
+      interface: NIC_INTERFACE_DEFAULT,
+      linked: true,
+    }
+
+    this.open = this.open.bind(this)
+    this.close = this.close.bind(this)
+    this.composeNic = this.composeNic.bind(this)
+    this.changeName = this.changeName.bind(this)
+    this.changeVnicProfile = this.changeVnicProfile.bind(this)
+    this.changeInterface = this.changeInterface.bind(this)
+    this.changeLinked = this.changeLinked.bind(this)
+    this.handleSave = this.handleSave.bind(this)
+  }
+
+  open (e) {
+    if (e && e.preventDefault) e.preventDefault()
```

Comment:
Is this necessary check first argument? Because I don't any other cases of using this method.

Reply:
Since `NicEditor` takes a `trigger` and clones it to add an `onClick` .... `{React.cloneElement(trigger, { onClick: this.open })}` .... having `open` be able to prevent an `<a href='#'>...</a>` from changing the browser URL is worth having this line.

But in the strictest view, no is it not absolutely necessary.  It just provides a slightly nice user experience.

Reply:
I mean not `preventDefault`, but `if (e && e.preventDefault)`

Reply:
Ah ok, now I understand the question.  I think I was thinking about a form field change handler and click hanlders together...and that is why code reviews are a good thing!

The check shouldn't be necessary as the onClick is a standard event etc etc.  I'll keep the `preventDefault()` in there.